### ISSUE TITLE
Compatible with margin loss and label smoothing

### ISSUE THREAD

**Issue by milliema** [hyperconnect/LADE#11]
Thanks for the impressive work!
I'd like to know whether the proposed method is compatible with margin loss (e.g. arc margin, cos margin, etc) and label smoothing? If yes, in which stage can we insert these techniques?

**Comment by juice500ml** [comment]
Thank you for the interesting idea you've brought here!
For example, for the ArcFace loss, s * cos \theta becomes the logits so that we feed them to the softmax, right?
PC Softmax can be applied just before applying the softmax function at inference, and LADER regularizes the logits, so I'm guessing that we can apply LADER to the margin losses too.

**Comment by milliema** [comment]
@juice500ml Thanks for the reply!
So the pipeline is: 
Image--> Feature--> Logits-->Logits with margins --> apply PC Softmax&LADER, is my understanding right?

**Comment by juice500ml** [comment]
Yes!
